rest-server/src/controllers/user.js

```diff
@@ -18,63 +18,49 @@
 // module dependencies
 const userModel = require('../models/user');
 const logger = require('../config/logger');
+const createError = require('../util/error');
 
 /**
  * Create / update a user.
  */
-const update = (req, res) => {
+const update = (req, res, next) => {
   const username = req.body.username;
   const password = req.body.password;
   const admin = req.body.admin;
   const modify = req.body.modify;
   if (req.user.admin ||
       username === req.user.username &&
       (typeof admin === 'undefined' || !admin) && modify) {
-    userModel.update(username, password, admin, modify, (err, state) => {
-      if (err || !state) {
-        logger.warn('update user %s failed', username);
-        return res.status(500).json({
-          error: 'UpdateFailed',
-          message: 'update user failed',
-        });
+    userModel.update(username, password, admin, modify, (err) => {
+      if (err) {
+        return next(createError.unknown(err));
       } else {
         return res.status(201).json({
           message: 'update successfully',
         });
       }
     });
   } else {
-    return res.status(401).json({
-      error: 'NotAuthorized',
-      message: 'not authorized',
-    });
+    next(createError('Forbidden', 'ERR_FORBIDDEN_USER', `Non-admin is not allow to do this operation.`));
   }
 };
 
 /**
  * Remove a user.
  */
-const remove = (req, res) => {
+const remove = (req, res, next) => {
   const username = req.body.username;
   if (req.user.admin) {
-    userModel.remove(username, (err, state) => {
-      if (err || !state) {
-        logger.warn('remove user %s failed', username);
-        return res.status(500).json({
-          error: 'RemoveFailed',
-          message: 'remove failed',
-        });
-      } else {
-        return res.status(200).json({
-          message: 'remove successfully',
-        });
+    userModel.remove(username, (err) => {
+      if (err) {
+        return next(createError.unknown(err));
```

Comment:
Is it 'internal server error'?

Reply:
`createError.unknown` will not transform known error objects like

https://github.com/Microsoft/pai/blob/qixcheng/rest-server-refine-error/user/rest-server/src/models/user.js#L90

or 

https://github.com/Microsoft/pai/blob/qixcheng/rest-server-refine-error/user/rest-server/src/models/user.js#L97

It will only transform unknown upstream errors (usually from HDFS/etcd utilities) into general 500 errors.

At present, most of `UnknownError`s are unhandled `UpstreamServerError`s. Like other non-`UpstreamServerError` `UnknownError`s, it can only be resolved manually by developers. So when any `UnknownError` happened, report `error.message` to developers may be the only solution, no matter it is from upstream server or rest-server itself. So I think different `error.message` in `UnknownError` is helpful enough.